autosubmit/job/job.py

```diff
@@ -1429,13 +1383,12 @@ def update_status(self, as_conf: AutosubmitConfig, failed_file: bool = False) ->
         :param failed_file: boolean, if True, checks if the job failed
         :return:
         """
-        previous_status = self.status
-        self.prev_status = previous_status
+        self.prev_status = self.status
         new_status = self.new_status
         if new_status == Status.COMPLETED:
             Log.debug(
                 "{0} job seems to have completed: checking...".format(self.name))
-            self._platform.get_completed_files(self.name, wrapper_failed=self.packed)
+            self._platform.get_completed_files(self.name, wrapper_failed=self.packed) # type: ignore
```

Comment:
Two spaces before a `#`, or it adds a lint warning.

Reply:
And the `self._platform` shouldn't be `None`. Maybe define the `self._platform` as `Optional['Platform']` and then set the ignore here to the error complained by mypy

In the output of mypy, it shows the category of the type error, which using `Optional` results in `union-attr` (the union type `None` doesn't have the function `.get_completed_files`, but we know it's not `None` at this point).

So we can reduce the scope of the ignore: `  # ignore[union-attr]`. That should raise other issues if we introduce any type bugs, but will silence mypy for this line.